### PR TITLE
GDML material input

### PR DESCRIPTION
**Pull request by bam241**
This adds the capability to write material input deck.
Documentation about GDML material input can be found [[here](http://lcgapp.cern.ch/project/simu/framework/GDML/doc/GDMLmanual.pdf)], p12-13.

Material input in GDML should have 3 sections:
`<isotope> </isotope>`
`<element></element>`
`<material></material>`
to avoid redundancy within the isotopes declaration I did't include the part writing the isotopes in the material.
I was thinking about adding the capability in `nucname.cpp` and/or in MaterialLibrary (see #1104 )

This also adds standard chemical element names from [wikipedia](https://en.wikipedia.org/wiki/List_of_chemical_elements), allowing to name the different element present int the material.

As the element seems to be required to build a material in GDML format, this loop over the material composition, group isotopes by elements, and build the elements with the proper abundances.
It then build a GDML material using the different elements previously build.

To build the elements input, the standard `nucname::name` are used for the isotopes (i.e.: `Am241`).
To avoid conflict if other material have the same elements (with or without the same abundances) the elements are name as: `MATNAME_ELEMENTNAME`.

To be done:
- [x] python binding
- [x] nosetests
- [ ] Isotopes GDML input deck from ??

### CODE REVIEW

File: pyne/material.pyx [hunk 1]
```diff
@@ -345,6 +345,19 @@ cdef class _Material:
         cdef std_string card
         card = self.mat_pointer.mcnp(frac_type.encode())
         return card.decode()
+    
+    def gdml(self):
```
Comment (zxkjack123) [inline pyne/material.pyx:375]: Docstring of this function is not clear. 
- Parameters are usually before Return.
- There isn't an input paremeter, what's the `int 0`?

Comment (bam241) [reply pyne/material.pyx:375]: sorry, I forgot the clean the copy paste :)

thx for the catch!

**Comment by bam241** [comment]
GDML needs the isotopes definition in the input deck, but to avoid redundancy I am not sure it should occur at the material level...

**Comment by bam241** [comment]
@gonuke this should be ready for a round of reviews.

File: src/material.cpp [hunk 2]
```diff
@@ -961,6 +961,79 @@ std::string pyne::Material::mcnp_frac(std::map<int, double> fracs, std::string f
...
+    mat_name = metadata["name"].asString();
+  }
+  if (metadata.isMember("mat_number")) {
+    mat_name_by_num += std::to_string(metadata["mat_number"].asInt());
+  }
+
+  // 'Isotope list'
+
+  // 'Element list'
+  std::map<int, double> fracs = comp;
```
Comment (gonuke) [inline src/material.cpp:980]: I don't think this is used.

File: src/material.cpp [hunk 3]
```diff
@@ -961,6 +961,79 @@ std::string pyne::Material::mcnp_frac(std::map<int, double> fracs, std::string f
...
+  // loop over the composition and group isotopes per element
+  for (auto it : comp) {
+    int z = nucname::znum(it.first);
+    std::string element_name = mat_name_by_num + "_" + nucname::name_elt[nucname::zz_name[z]];
+    if (elements_list.find(z) == elements_list.end()) {
+      std::map<int, double> list;
+      list[it.first] = it.second;
+      elements_list.insert(std::pair<int, std::map<int, double>>(z, list));
+
+      // add fraction in the material composition
```
Comment (gonuke) [inline src/material.cpp:992]: why use `element_name` here instead of `z`?

File: src/material.cpp [hunk 4]
```diff
@@ -961,6 +961,79 @@ std::string pyne::Material::mcnp_frac(std::map<int, double> fracs, std::string f
...
+      element_comp.insert( std::make_pair(element_name, it.second));
+    } else {
+      elements_list[z][it.first] = it.second;
+      // add fraction in the material composition
+      element_comp[element_name] += it.second;
+    }
+  }
+
+  // loop over the different chemical elements
+  for (auto it : elements_list) {
```
Comment (gonuke) [inline src/material.cpp:1002]: would it be more readable to use `nuclide` instead of `it`?

File: src/nucname.cpp [hunk 5]
```diff
@@ -6,8 +6,148 @@
...
+  lname["Be"] = "Beryllium";
+  lname["B"] = "Boron";
+  lname["C"] = "Carbon";
+  lname["N"] = "Nitrogen";
+  lname["O"] = "Oxygen";
+  lname["F"] = "Fluorine";
+  lname["Ne"] = "Neon";
+  lname["Na"] = "Sodium";
+  lname["Mg"] = "Magnesium";
+  lname["Al"] = "Aluminium";
```
Comment (gonuke) [inline src/nucname.cpp:28]: What is the basis for these spellings.  Are they based on something from Geant?  They may not be universal....

**Review by gonuke — CHANGES_REQUESTED** [review]
A few little changes, but seems mostly OK

**Pull request closed without merging**